### PR TITLE
[Intersections_2] Make the crossing point computation deterministic

### PR DESCRIPTION
**Pull request by sloriot**
Always use the same ordered segments

### CODE REVIEW

File: Intersections_2/include/CGAL/Intersections_2/Segment_2_Segment_2.h [hunk 1]
```diff
@@ -215,8 +242,8 @@ do_intersect_with_info(const typename K::Segment_2 &seg1,
     typename K::Compare_xy_2 compare_xy;
 
   // first try to filter using the bbox of the segments
-    if (less_xy(A2,B1)
-     || less_xy(B2,A1))
+    if (certainly(less_xy(A2,B1))
```
Comment (sloriot) [inline Intersections_2/include/CGAL/Intersections_2/Segment_2_Segment_2.h:245]: @afabri this change is not related to the patch but I fixed it while I was in this file. Do you agree that it does not make sense to have a filtered failure in that case?

Comment (afabri) [reply Intersections_2/include/CGAL/Intersections_2/Segment_2_Segment_2.h:245]: Latest the `switch(make_certain(..))` will lead to the exception being thown, so we can throw it here as well.

**Comment by lrineau** [comment]
@sloriot During the rebase of this PR, I could not rebase it on `5.3.x-branch`, because of important conflicts. I have rebased it on `5.4.x-branch`. I let you rebase it on `5.3.x-branch` if you think that is important.

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.5-Ic-25 (5.4 is fine)

**Pull request merged**